pkg/ndctl/region.go

```diff
@@ -184,16 +190,6 @@ func (r *region) CreateNamespace(opts CreateNamespaceOpts) (Namespace, error) {
 		}
 	}
 
-	if opts.Size != 0 {
-		available := r.MaxAvailableExtent()
-		if available == uint64(C.ULLONG_MAX) {
-			available = r.AvailableSize()
-		}
-		if opts.Size > available {
-			return nil, fmt.Errorf("create namespace with size %v: %w", opts.Size, pmemerr.NotEnoughSpace)
-		}
-	}
-
 	align := defaultAlign
```

Comment:
The `defaultAlign` variable serves no purpose and just makes the code harder to read. Please remove it.

Reply:
opts.Align is unused. We can remove it and the dead code checking it:
```
diff --git a/pkg/ndctl/ndctl.go b/pkg/ndctl/ndctl.go
index 9545c6e1a..cbaac22f3 100644
--- a/pkg/ndctl/ndctl.go
+++ b/pkg/ndctl/ndctl.go
@@ -29,7 +29,6 @@ type CreateNamespaceOpts struct {
 	Name       string
 	Size       uint64
 	SectorSize uint64
-	Align      uint64
 	Type       NamespaceType
 	Mode       NamespaceMode
 	Location   MapLocation
diff --git a/pkg/ndctl/region.go b/pkg/ndctl/region.go
index abc714d36..990122527 100644
--- a/pkg/ndctl/region.go
+++ b/pkg/ndctl/region.go
@@ -150,7 +150,6 @@ func (r *region) GetAlign() uint64 {
 }
 
 func (r *region) CreateNamespace(opts CreateNamespaceOpts) (Namespace, error) {
-	defaultAlign := mib2
 	var err error
 	/* Set defaults */
 	if opts.Type == "" {
@@ -190,19 +189,7 @@ func (r *region) CreateNamespace(opts CreateNamespaceOpts) (Namespace, error) {
 		}
 	}
 
-	align := defaultAlign
-	if opts.Align != 0 {
-		if opts.Mode == SectorMode || opts.Mode == RawMode {
-			klog.V(4).Infof("%s mode does not support setting an alignment, hence ignoring alignment", opts.Mode)
-		} else {
-			var err error
-			align, err = r.AdaptAlign(opts.Align)
-			if err != nil {
-				return nil, err
-			}
-		}
-	}
-
+	align := mib2
 	if opts.Size != 0 {
 		namespacealign := align * r.InterleaveWays()
 		regionalign := r.GetAlign()
```

Reply:
I was not sure do you mean stop using it, or stop and remove it from Opts. I was thinking, perhaps we keep it so that it can possibly used again in the future? BTW we have also other fields in Opts that we don't use or always set to one default, like SectorSize. Type, Mode, Location . Although Sector mode may see it's comeback based on another discussion.

Reply:
I understand that conceptually `pkg/ndctl` is allowed to support more than we need. But this particular code has no test coverage, therefore I prefer to remove it. If it turns out that we need it, we bring it back and then ensure that it really works.

If we keep it in place without testing, we might not remember that it is potentially broken when we start to use it.